src/Marten/Schema/SchemaBuilder.cs

```diff
@@ -22,7 +22,7 @@ public static void WriteSchemaObjects(DocumentMapping mapping, IDocumentSchema s
             var args = new List<UpsertArgument>
             {
                 new UpsertArgument {Arg = "docId", PostgresType = pgIdType},
-                new UpsertArgument {Arg = "doc", PostgresType = "JSON"}
+                new UpsertArgument {Arg = "doc", PostgresType = "JSONB"}
```

Comment:
9.4 yells at you if you don't specify JSONB here and in CommandExtensions (See below)